app/main/plugins/hain-plugin-filesearch/shared-util.js

```diff
@@ -14,10 +14,21 @@ function injectEnvVariable(dirPath) {
   }
 
   // Inject Environment Variables
-  for (const envVar in process.env) {
-    const value = process.env[envVar];
-    _path = _path.replace(`\${${envVar}}`, value);
+  let tokens = dirPath.match(/\${.*?}/g);
+
+  if (tokens) {
```

Comment:
I can see this would speed things up, but let's be sure tokens are being replaced properly.

Reply:
I'll look at adding some tests - it's complicated by the fact that this function uses process.env directly, so I need a way of mocking this object. I've seen https://www.npmjs.com/package/mock-env , do you have a different idea for how to mock process.env in a test?